File: lab1/node-app/app-basic.js
const express = require('express');
const app = express();
const port = 3001;

//Вычисление необходимых данных
const dateNow = new Date();
const weekDay = dateNow.getDate();
let weekDayText = '';

if(weekDay == 0){
    weekDayText = 'Воскресенье';
}
if(weekDay == 1){
    weekDayText = 'Понедельник';
}
if(weekDay == 2){
    weekDayText = 'Вторник';
}
if(weekDay == 3){
    weekDayText = 'Среда';
}
if(weekDay == 4){
    weekDayText = 'Четверг';
}
if(weekDay == 5){
    weekDayText = 'Пятница';
}
if(weekDay == 6){
    weekDayText = 'Суббота';
}


// 1. Текстовый эндпоинт
app.get('/', (req, res) => {
    res.send('Hello, world!');
});

// 2. JSON эндпоинт 
app.get('/api/date', (req, res) => {
    res.json({date: `${dateNow.getFullYear()}-${dateNow.getMonth()+1}-${dateNow.getDate()}`, day : weekDayText});
});

app.listen(port, () => {
    console.log(`Сервер запущен на http://localhost:${port}`);
});

File: lab1/flask-app-extra/app-basic.py
from flask import Flask, jsonify, request
import time

app = Flask(__name__)

# 1. Текстовый эндпоинт
@app.route('/')
def home():
    return 'Hello, world!'

# 2. JSON эндпоинт 1
@app.route('/api/date')
def status():
    return jsonify({"date": "2024-01-01", "day": "Monday"})

if __name__ == '__main__':
    app.run(port=5001, debug=True)